backend/src/main/java/com/htecgroup/skynest/service/impl/FileServiceImpl.java

```diff
@@ -126,29 +155,49 @@ private List<FileResponse> asFileResponseList(List<FileMetadataEntity> allFiles)
         .collect(Collectors.toList());
   }
 
-  private FileMetadataEntity initFileMetadata(String name, long size, String type, UUID bucketId) {
+  private FileMetadataEntity initFileMetadata(
+      String name, long size, String type, UUID bucketId, UUID folderId) {
+
+    if (bucketId != null && folderId != null) throw new IllegalArgumentException();
 
     UserEntity currentUserEntity =
         userRepository
             .findById(currentUserService.getLoggedUser().getUuid())
             .orElseThrow(UserNotFoundException::new);
 
-    BucketEntity bucketEntity =
-        bucketRepository.findById(bucketId).orElseThrow(BucketNotFoundException::new);
-
     FileMetadataEntity fileMetadataEntity = new FileMetadataEntity();
 
     fileMetadataEntity.setCreatedBy(currentUserEntity);
     fileMetadataEntity.setName(name);
 
-    fileMetadataEntity.setBucket(bucketEntity);
-    fileMetadataEntity.setParentFolder(null);
     fileMetadataEntity.setSize(size);
     fileMetadataEntity.setType(type);
 
+    BucketEntity bucket;
+    FolderEntity parentFolder;
+    if (bucketId != null) {
+      parentFolder = null;
+      bucket = bucketRepository.findById(bucketId).orElseThrow(BucketNotFoundException::new);
+    } else {
+      parentFolder = folderRepository.findById(folderId).orElseThrow(FolderNotFoundException::new);
+      bucket = parentFolder.getBucket();
+    }
+    fileMetadataEntity.setBucket(bucket);
+    fileMetadataEntity.setParentFolder(parentFolder);
+
     return fileMetadataEntity;
   }
 
+  private FileMetadataEntity initFileMetadataWithBucket(
+      String name, long size, String type, UUID bucketId) {
```

Comment:
Why not put this  `bucketRepository.findById(bucketId).orElseThrow(BucketNotFoundException::new);`
here and the same corresponding one in the other method
Also this two lines will be moved into _**initFileMetadata**_
```
 fileMetadataEntity.setBucket(bucket);
    fileMetadataEntity.setParentFolder(parentFolder);
```

Reply:
makes sense